service/src/main/java/uk/gov/hmcts/reform/fpl/service/OrganisationService.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.fpl.config.LocalAuthorityUserLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.exceptions.UnknownLocalAuthorityCodeException;
+import uk.gov.hmcts.reform.rd.client.OrganisationApi;
+import uk.gov.hmcts.reform.rd.model.Status;
+import uk.gov.hmcts.reform.rd.model.User;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Slf4j
+public class OrganisationService {
+    private final LocalAuthorityUserLookupConfiguration localAuthorityUserLookupConfiguration;
+    private final OrganisationApi organisationApi;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    @Autowired
+    public OrganisationService(LocalAuthorityUserLookupConfiguration localAuthorityUserLookupConfiguration,
+                               OrganisationApi organisationApi,
+                               AuthTokenGenerator authTokenGenerator) {
+        this.localAuthorityUserLookupConfiguration = localAuthorityUserLookupConfiguration;
+        this.organisationApi = organisationApi;
+        this.authTokenGenerator = authTokenGenerator;
+    }
+
+
+    public List<String> findUserIdsInSameOrganisation(String authorisation, String userId, String localAuthorityCode) {
+        List<String> userIds = new ArrayList<>();
+
+        try {
+            addUsersFromSameOrganisationBasedOnAppConfig(localAuthorityCode, userIds);
+        } catch (UnknownLocalAuthorityCodeException ex) {
+            try {
+                addUsersFromSameOrganisationBasedOnReferenceData(authorisation, userIds);
+            } catch (Exception e) {
+                log.warn("Can't find LocalAuthority for code: "
+                    + localAuthorityCode
+                    + " in app config and the PRD endpoint call threw an exception.", e);
+            }
+        }
+
+        if (!userIds.contains(userId)) {
+            userIds.add(userId);
+        }
+
+        return userIds;
+    }
+
+    private void addUsersFromSameOrganisationBasedOnAppConfig(String localAuthorityCode, List<String> userIds) {
```

Comment:
Not big fan of methods like that. IMO better and cleaner is to append users in main method with smaller methods that return a list to be appended e.g.

```
private void getUsersFromSameOrganisationBasedOnAppConfig(String localAuthorityCode) { // if you need to rename it for more clarity
  return localAuthorityUserLookupConfiguration.getUserIds(localAuthorityCode)
}

private void getUsersFromSameOrganisationBasedOnReferenceData(String authorisation) {
         return organisationApi
             .findUsersByOrganisation(authorisation, authTokenGenerator.generate(), Status.ACTIVE)
             .stream()
             .map(User::getUserIdentifier)
     }
 }
```

Reply:
Agree, will reefactor.

Reply:
Refactored.